sklearn/model_selection/_search.py

```diff
@@ -751,16 +769,31 @@ def evaluate_candidates(candidate_params):
                                      .format(n_splits,
                                              len(out) // n_candidates))
 
+                # For callabe self.scoring, the return type is only know after
+                # calling. If the return type is a dictionary, the error scores
+                # can now be inserted with the correct key.
```

Comment:
I would add in the comment that the type checking of out will happen in _insert_error_scores